### ISSUE TITLE
[Feature Request] Authentication logging

### ISSUE THREAD

**Issue by mtanml** [AzureAD/microsoft-identity-web#821]
Is logging available?   If so, can you please point me to the documentation?

I'm looking for something like the WithLogging parameter for Microsoft Identity Client:

https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.client.abstractapplicationbuilder-1.withlogging?view=azure-dotnet

Thanks

**Comment by jennyf19** [comment]
@mtanml we integrate with the asp net core logs, which you can set up in the [appsettings](https://github.com/AzureAD/microsoft-identity-web/blob/master/tests/WebAppCallsWebApiCallsGraph/Client/appsettings.json#L31), more [information here](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0) and more on [configuring logging](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0#configure-logging).

**Comment by mtanml** [comment]
@jennyf19 thanks for the info!   I was able to enable logging, but is there a way to limit the logs to authentication info?   And I'm not seeing the level of detail that we get with Microsoft Identity Client (token acquisition/exchange, etc.)   We use this info to troubleshoot user logins.   Thanks!

**Comment by jennyf19** [comment]
@mtanml I understand. we do not hook up to the MSAL logs. So, yes, that would be a feature request.
cc: @jmprieur

**Comment by jennyf19** [comment]
@mtanml i have a branch w/ an [initial commit](https://github.com/AzureAD/microsoft-identity-web/tree/jennyf/logs) to enable the MSAL .NET logs, but if you have suggestions or feedback on the developer experience, that would be much appreciated. 

In appsettings.json, you would need to do the following in the `AzureAD` section:
```
 "EnableMsalPiiLogs": true, // set to false by default, so not necessary
 "LogLevel": "Info" //MSAL .NET includes "Info" "Warning" "Verbose" and "Error" logging options
```

I did the following to see the MSAL .NET logs in the console, but i'm sure you have something more interesting:
```
"Logging": {
        "LogLevel": {
            "Default": "Information",
            "Microsoft": "Warning",
            "Microsoft.Hosting.Lifetime": "Information"
        },
        "Console": {
            "LogLevel": {
                "Default": "Information"
            }
        }
```

cc: @jmprieur

**Comment by mtanml** [comment]
@jennyf19 I won't have time to do any testing soon, but I **really** appreciate the super-fast work on this!   In our project that implements Microsoft.Identity.Client we use a method with the following parameters to capture authentication log info and write it to our database.   As long as we'd be able to do the same in Microsoft.Identity.Web, then you have met our needs.   Thanks a million!

private static void Log(Microsoft.Identity.Client.LogLevel level, string message, bool containsPii)

**Comment by jennyf19** [comment]
@mtanml we've changed a few things from the initial branch, so here is the info on [logging](https://github.com/AzureAD/microsoft-identity-web/wiki/Logging) in the wiki and we'll get this out in the next release.

**Comment by mtanml** [comment]
Excellent, thank you @jennyf19  !

**Comment by jennyf19** [comment]
@mtanml we will get this in 1.4.1 release, which will probably be tomorrow (12.15)

**Comment by jennyf19** [comment]
@mtanml Included in [1.4.1 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.4.1)

**Comment by mtanml** [comment]
@jennyf19 I was just about to write you... works like a charm  :)  

Thanks again!
